webdriver/tests/bidi/network/add_intercept/url_patterns.py

```diff
@@ -0,0 +1,238 @@
+        ("https://{wpt_host}/", "https://{wpt_host}/#ref"),
+        ("https://{wpt_host}/#ref1", "https://{wpt_host}/#ref2"),
+    ],
+)
+async def test_string_patterns_matching(
+    bidi_session,
+    wait_for_event,
+    url,
+    setup_network_test,
+    fetch,
+    substitute_host,
+    pattern,
+    url_template,
+):
+    network_events = await setup_network_test(
+        events=[
+            "network.beforeRequestSent",
+        ]
+    )
+
+    intercept = await bidi_session.network.add_intercept(
+        phases=["beforeRequestSent"],
+        url_patterns=[{"type": "string", "pattern": substitute_host(pattern)}],
+    )
+
+    on_network_event = wait_for_event("network.beforeRequestSent")
+    asyncio.ensure_future(fetch(substitute_host(url_template)))
+    event = await on_network_event
+
+    assert_before_request_sent_event(event, is_blocked=True, intercepts=[intercept])
+
+    await bidi_session.network.remove_intercept(intercept=intercept)
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "pattern, url_template",
+    [
+        ("https://{wpt_host}/", "https://some.other.host/"),
+        ("https://{wpt_host}:1234/", "https://{wpt_host}:5678/"),
+        ("https://{wpt_host}/", "https://{wpt_host}:5678/"),
+        ("https://{wpt_host}/path", "https://{wpt_host}/other/path"),
+        ("https://{wpt_host}/path", "https://{wpt_host}/path/continued"),
+        ("https://{wpt_host}/pathcase", "https://{wpt_host}/PATHCASE"),
+        ("https://{wpt_host}/?searchcase", "https://{wpt_host}/?SEARCHCASE"),
+        ("https://{wpt_host}/?key", "https://{wpt_host}/?otherkey"),
+        ("https://{wpt_host}/?key", "https://{wpt_host}/?key=value"),
+        ("https://{wpt_host}/?a=b&c=d", "https://{wpt_host}/?c=d&a=b"),
+        ("https://{wpt_host}/??", "https://{wpt_host}/?"),
+    ],
+)
+async def test_string_patterns_not_matching(
+    bidi_session,
+    wait_for_event,
+    setup_network_test,
+    fetch,
+    substitute_host,
+    pattern,
+    url_template,
+):
+    network_events = await setup_network_test(
+        events=[
+            "network.beforeRequestSent",
+        ]
+    )
+
+    intercept = await bidi_session.network.add_intercept(
+        phases=["beforeRequestSent"],
+        url_patterns=[{"type": "string", "pattern": substitute_host(pattern)}],
+    )
+
+    on_network_event = wait_for_event("network.beforeRequestSent")
+    asyncio.ensure_future(fetch(substitute_host(url_template)))
+    event = await on_network_event
+
+    assert_before_request_sent_event(event, is_blocked=False)
+
+    await bidi_session.network.remove_intercept(intercept=intercept)
+
+
+@pytest.fixture
+def substitute_host(server_config):
+    """This test will perform various requests which should not reach the
+    external network. All strings refering to a domain will define it as a
+    placeholder which needs to be dynamically replaced by a value from the
+    current server configuration"""
+
+    def _substitute_host(str):
```

Comment:
Even if you name this as `substitue_host` (and update the `return` accordingly), it should still work.

This pattern seems to be common in pytest.

Reply:
Right, will update!